### ISSUE TITLE
Adding an int and a float is incorrectly allowed

### ISSUE THREAD

**Issue by umanwizard** [rust-analyzer/rust-analyzer#1609]
No issue is reported for the following code (which `rustc` refuses).

```rust
fn main() {
    let i = 100;
    let f = 100.0;
    let foo = i + f;
}
```

Furthermore, `i` here is incorrectly deduced to be `f64`.

**Comment by flodiebold** [comment]
We don't report any type errors at all so far. Or do you mean you're not seeing an error from the `cargo watch` integration either?

**Comment by umanwizard** [comment]
Thank you. I wasn’t aware that error reporting is not a feature yet.

**Comment by bjorn3** [comment]
> I wasn’t aware that error reporting is not a feature yet.

It does support errors, but only syntax errors and name resolution errors.
